### PR TITLE
feat(kit): `InputTag` add max number of rows for `expandable` mode

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

related #3265

## What is the new behavior?

new input `rows` added, it limits max number of textfield rows in expandable mode, same as in `Textarea`

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/components/input-tag/input-tag.component.ts [hunk 1]
```diff
@@ -537,4 +554,10 @@ export class TuiInputTagComponent
     private clippedValue(value: string): string {
         return value.slice(0, this.maxLength || value.length);
     }
+
+    private get lineHeight(): number {
+        return this.labelOutside
+            ? TAG_SIZE_REM[this.controller.size] + 2 * 0.125
```
Comment (splincode) [inline projects/kit/components/input-tag/input-tag.component.ts:560]: Magic number?
`2 * 0.125` maybe move it to named constant?

**Review by splincode — APPROVED** [review]

File: projects/kit/components/input-tag/input-tag.component.ts [hunk 2]
```diff
@@ -127,6 +138,9 @@ export class TuiInputTagComponent
     @tuiDefaultProp()
     expandable = true;
 
+    @Input()
+    rows = Infinity;
```
Comment (waterplea) [inline projects/kit/components/input-tag/input-tag.component.ts:142]: I wonder if we can deprecate `expandable` in favor of this input with 1 === expandable: false.

**Review by waterplea — COMMENTED** [review]
When you type a tag and hit "Enter" it scrolls to top.

File: projects/demo/src/modules/components/input-tag/examples/8/index.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<tui-input-tag
+    class="input"
+    [tuiTextfieldLabelOutside]="true"
+    [formControl]="control"
+    [rows]="3"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/input-tag/examples/8/index.html:5]: Is this scroll expected ?

(video shows what happens when i type value and press "enter")

https://user-images.githubusercontent.com/35179038/210048874-1a6c9673-bb92-47d5-b386-8ccd30cf06c1.mov

File: projects/demo/src/modules/components/input-tag/input-tag.component.ts [hunk 4]
```diff
@@ -70,6 +70,11 @@ export class ExampleTuiInputTagComponent extends AbstractExampleTuiControl {
         HTML: import('./examples/7/index.html?raw'),
     };
 
+    readonly example8: TuiDocExample = {
+        TypeScript: import('./examples/8/index.ts?raw'),
+        HTML: import('./examples/8/index.html?raw'),
+    };
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/input-tag/input-tag.component.ts:76]: less file

File: projects/kit/components/input-tag/input-tag.component.ts (hunk 2 above)
Comment (vladimirpotekhin) [reply projects/kit/components/input-tag/input-tag.component.ts:143]: Scrolling fixed. 
For now I can't deprecate `expandable` due to backward compatibility. I added TODO for 4.0

**Review by nsbarsukov — APPROVED** [review]

File: projects/kit/components/input-tag/input-tag.style.less [hunk 5]
```diff
@@ -28,13 +28,12 @@
...
 .t-scrollbar {
     margin: 0 -0.25rem;
-    padding: 0 0.25rem;
 }
 
 .t-tags {
     display: flex;
-    align-items: center;
-    box-sizing: border-box;
+    padding-inline-start: 0.25rem;
```
Comment (splincode) [inline projects/kit/components/input-tag/input-tag.style.less:35]: image

drop safari 12?

Comment (vladimirpotekhin) [reply projects/kit/components/input-tag/input-tag.style.less:36]: duplicated with padding-right

**Pull request merged**